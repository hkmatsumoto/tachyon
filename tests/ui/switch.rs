fn switch(x: i32) -> i32 {
    match x {
        0 => 99,
        2 => 99 - 2,
        x => 100 - x,
    }
}
