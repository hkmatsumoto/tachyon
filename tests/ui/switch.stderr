; ModuleID = 'top'
source_filename = "top"

define i32 @switch(i32 %x) {
entry:
  %_0 = alloca i32, align 4
  %_2 = alloca i32, align 4
  %_3 = alloca i32, align 4
  br label %bb0

bb0:                                              ; preds = %entry
  switch i32 %x, label %bb1 [
    i32 0, label %bb3
    i32 2, label %bb2
  ]

bb1:                                              ; preds = %bb0
  store i32 %x, i32* %_2, align 4
  %0 = load i32, i32* %_2, align 4
  store i32 %0, i32* %_3, align 4
  %1 = load i32, i32* %_3, align 4
  %2 = sub i32 100, %1
  store i32 %2, i32* %_0, align 4
  br label %bb4

bb3:                                              ; preds = %bb0
  store i32 99, i32* %_0, align 4
  br label %bb4

bb2:                                              ; preds = %bb0
  store i32 97, i32* %_0, align 4
  br label %bb4

bb4:                                              ; preds = %bb3, %bb2, %bb1
  %3 = load i32, i32* %_0, align 4
  ret i32 %3
}
